keps/693-multikueue/README.md

```diff
@@ -0,0 +1,234 @@
+# KEP-693: MultiKueue
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+Introduce an new AdmissionCheck (called MultiKueue) with dedicated API
+and controller that will provide multi-cluster capabilities to Kueue.
+
+## Motivation
+Many of Kueue's users are running multiple clusters and would like to
+have a way to easily distribute batch jobs across them to keep all of
+them utilized. Without a global distribution point, some clusters may
+get less jobs they are able to process, while the others get more, leading
+to underutilization and higher costs. 
+
+### Goals
+* Allow Kueue distribute batch jobs across multiple clusters,
+while maintaining the specified quota limits.
+* Provide users with a single entry point through which the jobs
+can be submitted and monitored, just like they were running in
+a single cluster.
+* Be compatible with all Kueue's features (priorities, borrowing, preemptions, etc)
+and most of integrations.
+* Allow to upgrade single cluster Kueue deployments to multicluster without
+much hassle.
+
+### Non-Goals
+* Solve storage problem. It is assumed that the distributed jobs are
+either location-flexible (for a subset of clusters) or are copying the 
+data as a part of the startup process.
+* Synchronize configuration across the clusters. It is expected that the 
+user will create the appropriate objects, roles and permissions
+in the clusters (manually, using gitops or some 3rd-party tooling).
+* Set up authentication between clusters.
+
+## Proposal
+
+Introduce MultiKueue AdmissionCheck, controller and configuration API. 
+
+Establish the need for a designated management cluster.
+
+![Architecture](arch.png "Architecture")
+
+For each workload coming to a ClusterQueue (with the MultiKueue AdmissonCheck enabled)
+in the management cluster, and getting past the preadmission phase in the 
+two-phase admission process (meaning that the global quota is ok), 
+MultiKueue controller will create it in the defined worker clusters and wait 
+until some Kueue running there admits the workload. 
+Then it will remove the workloads from the remaining clusters and allow the
+single instance of the job to proceed. The workload will be also admitted in 
+the management cluster.
+
+There will be no job controllers running in the management clusters or they will be
```

Comment:
I think disabling them somehow (i.e. using annotations) is a better and more k8s-native approach and it would be easier to use MultiKueue in managed Kubernetes Services (by any Cloud provider) as Cloud providers offer limited to no access to Control Plane

Reply:
May, and then MK controller will unadmit/remove one of them. The job will not start, we start it only after there is only one admitted workload and the rest is deleted.

Reply:
I believe this is already explained in the paragraph above.